### PR TITLE
Accept no content-type as json

### LINKED ISSUE #253: Header `Content-type: application/json` should not be mandatory
When adding documents, if the header `Content-type: application/json` is not present, the `application/json` type should be used.

⚠️ Add tests

### PR DESCRIPTION
**Pull request by irevoire**
closes #253

### CODE REVIEW

File: meilisearch-http/src/routes/document.rs [hunk 1]
```diff
@@ -29,6 +30,19 @@ macro_rules! guard_content_type {
 }
 
 guard_content_type!(guard_json, "application/json");
+*/
+
+fn guard_json(head: &actix_web::dev::RequestHead) -> bool {
+            if let Some(content_type) = head.headers.get("Content-Type") {
+                content_type
+                    .to_str()
```
Comment (MarinPostma) [inline meilisearch-http/src/routes/document.rs:38]: This returns an option whether the bytes from the header could be converted to a str (if I'm not mistaken), so you are saying that is we cannot convert the bytes to str, we assume it's json

File: meilisearch-http/src/routes/document.rs [hunk 2]
```diff
@@ -29,6 +30,19 @@ macro_rules! guard_content_type {
...
+                    .map(|v| v.contains("application/json"))
+                    // if no content-type is specified we still accept the data as json!
+                    .unwrap_or(true)
+            } else {
+                false
```
Comment (MarinPostma) [inline meilisearch-http/src/routes/document.rs:43]: This is where you want to return true: there is no content type specified

**Review by MarinPostma — CHANGES_REQUESTED** [review]

File: meilisearch-http/src/routes/document.rs (hunk 2 above)
Comment (irevoire) [reply meilisearch-http/src/routes/document.rs:43]: Oh yes, you're right!
Strangely, it worked the other way around :grimacing: 
I added a patch. Thank you!

**Review by MarinPostma — APPROVED** [review]
good job :)

**Review by curquiza — APPROVED** [review]
Thanks @irevoire for the fix and the test addition 🙂

bors merge

**Pull request merged**